wfsim/strax_interface.py

```diff
@@ -60,34 +65,46 @@ def rand_instructions(c):
     return instructions
 
 
-@export
-def read_optical(file):
-    data = uproot.open(file)
-    all_ttrees = dict(data.allitems(filterclass=lambda cls: issubclass(cls, uproot.tree.TTreeMethods)))
-    e = all_ttrees[next(iter(all_ttrees))]
 
-    n_events = len(e.array('eventid'))
+def read_optical(file, c):
+
+    data = uproot.open(file)
+    if c['mc_version_above_4']:
+        e = data.get('events')
+    else:
+        all_ttrees = dict(data.allitems(filterclass=lambda cls: issubclass(cls, uproot.tree.TTreeMethods)))
+        e = all_ttrees[next(iter(all_ttrees))]
+
+    
+    n_events = len(e['eventid'].array(library="np"))
     # lets separate the events in time by a constant time difference
     time = np.arange(1, n_events+1)
 
     # Events should be in the TPC
-    xp = e.array("xp_pri") / 10
-    yp = e.array("yp_pri") / 10
-    zp = e.array("zp_pri") / 10
+    xp = e["xp_pri"].array(library="np") / 10
+    yp = e["yp_pri"].array(library="np") / 10
+    zp = e["zp_pri"].array(library="np") / 10
 
-    channels = e.array("pmthitID")
-    # TODO: It should be fixed following PMT mapping.
-    # In our MC, TPC PMT IDs are assigned at 10000-- -> 0--
-    # nVeto ones are assigned at 20000-- -> 2000--
-    channels = [[channel if channel < 20000 else channel - 18000 for channel in array] for array in channels]
-    timings = e.array("pmthitTime")*1e9
+    if c['nv']:
+        if c['mc_version_above_4']:
+            nV_pmt_id_offset = 2000
+        else:
+            nV_pmt_id_offset = 20000
+        channels = [[channel - nV_pmt_id_offset for channel in array] for array in e["pmthitID"].array(library="np")]
+        timings = e["pmthitTime"].array(library="np")*1e9
+    
+    else:
+        # TPC
+    
+        channels = e["pmthitID"].array(library="np")
+        timings = e["pmthitTime"].array(library="np")*1e9
 
     ins = np.zeros(n_events, dtype=instruction_dtype)
 
     ins['x'], ins['y'], ins['z'], ins['time'] = xp.flatten()  / 10, \
-                                                       yp.flatten() / 10, \
-                                                       zp.flatten() / 10, \
-                                                       1e7 * time.flatten()
+        yp.flatten() / 10, \
+        zp.flatten() / 10, \
```

Comment:
Duplicated division by 10 (see 20 lines above)?

Reply:
Yeah this snuck in at some point